### ISSUE TITLE
Do we need libsodium from source?

### ISSUE THREAD

**Issue by inetic** [clostra/newnode#57]
We have not customized it in any way and it's one of the things that slows down Travis the most.
Perhaps we can just install binaries?

**Comment by shalunov** [comment]
My recollection was that I added it as a submodule because you had to install it from source on Debian. The build worked fine for Greg and me with sodium from brew, IIRC.

**Comment by inetic** [comment]
I do remember a little debate about this, but I'm as well foggy about the details. I think the problem I had was Ubuntu not having the 2.1 version of `libevent`. We fixed it by adding `libevent` into our sources as a submodule and generalized this approach to `libsodium`. It wasn't a big issue as we needed to compile it only once because we didn't use Travis at the time.

On my current Ubuntu 16.4 I can get `libsodium` from packages and it works. The Ubuntu/precise(12.4) on Travis doesn't have it in its standard repositories but it can be added from [here](https://launchpad.net/~chris-lea/+archive/ubuntu/libsodium/+packages).

MacOS as you mentioned works too and it reduces the MacOS build/test time from ~11 minutes to ~6 minutes.
